test/e2e/aks.go

```diff
@@ -303,3 +310,48 @@ func GetWorkingAKSKubernetesVersion(ctx context.Context, subscriptionID, locatio
 
 	return maxVersion, nil
 }
+
+// GetLatestStableAKSKubernetesVersion returns the latest stable available Kubernetes version of AKS.
+func GetLatestStableAKSKubernetesVersion(ctx context.Context, subscriptionID, location, version string) (string, error) {
+	settings, err := auth.GetSettingsFromEnvironment()
+	if err != nil {
+		return "", err
+	}
+	authorizer, err := settings.GetAuthorizer()
+	if err != nil {
+		return "", err
+	}
+	containerServiceClient := containerservice.NewContainerServicesClient(subscriptionID)
+	containerServiceClient.Authorizer = authorizer
+	result, err := containerServiceClient.ListOrchestrators(ctx, location, ManagedClustersResourceType)
+	if err != nil {
+		return "", err
+	}
+
+	var orchestratorversions []string
+	var isaksversioninpreview []*bool
+	var foundWorkingVersion bool
+	var orchVersion string
+	var maxVersion string
+
+	for _, o := range *result.Orchestrators {
+		orchestratorversions = append(orchestratorversions, *o.OrchestratorVersion)
```

Comment:
here could we just skip the ones where `o.IsPreview` is true and not even add them to `orchestratorversions`?

Reply:
Sure, I have added this change. Thank you.